### ISSUE TITLE
Spurious "Your changes will be lost" warning message

### ISSUE THREAD

**Issue by matterhorn103** [OpenChemistry/avogadrolibs#1206]
**Avogadro version: (please complete the following information from the About box):**
- Flatpak of version 1.97.0 
- Avogadrolibs: 1.97.0
 - Qt: 5.15.7

**Desktop version: (please complete the following information):**
 - OS: OpenSUSE Tumbleweed but behaviour on Windows 10 is identical
 - Version 20221201

**Describe the bug**
If the user has an open molecule that is unsaved and then clicks the "Open..." button, a warning message appears:
```
Do you want to save the changes to the document?
Your changes will be lost if you don't save them.
Save | Discard | Cancel
```
And this is simply not true, if `Discard` is chosen and a file is then opened, the unsaved molecule is still there and the opened file is put into a new "molecule" in the Molecules window.

**Comment by ghutchis** [comment]
Yeah, there's something happening to the layers that's noted as a change. I haven't been able to find the exact bug.

**Comment by matterhorn103** [comment]
Can this be reopened? The spurious message still appears.

**Comment by ghutchis** [comment]
Please open a new bug report.
